### PR TITLE
Milestone: Add devise

### PR DESCRIPTION
**Pull request by Roseokpe**
In this project,  I added the [devise](https://github.com/heartcombo/devise) gem to the app and use it for the registration and login of users.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <%= f.label :name %><br />
+    <%= f.text_field :name %>
+  </div>
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %>
```
Comment (codepantha) [inline app/views/devise/registrations/new.html.erb:15]: - [x]  Please modify the registration view in devise to match the wireframes and their styling.
## Wireframe Guidelines
image

## Your Implementation
image

- [ ] An error 🐛 occurs when registering a new user. As a result, I am unable to test the login, email confirmation, and password recovery functionality of your app. This error has to do with the `post_counter` column in your users table. Please in your migration, kindly assign it a default value of 0 in order to get rid of this error. Please try to fix it 🙏
image

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "current-password" %>
+  </div>
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field">
+      <%= f.check_box :remember_me %>
```
Comment (codepantha) [inline app/views/devise/sessions/new.html.erb]: - [ ] Please modify the sign_in view in devise to match the wireframe and its styling.
## Wireframe Guidelines
image
## Your Implementation
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @Roseokpe , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Used GitFlow

## Required Changes ♻️
_Check the comments under the review._

- [ ] Add a License section to your README, and add a MIT license because it is important that every software (opensource included) has a license 👍 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Roseokpe, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
- [ ] As the previous reviewer suggested kindly add a license to your project, you can choose MIT or any other open-source license
- [ ] As the previous reviewer mentioned your sign-in and sign-up pages should look as the provided wireframe, kindly implement this so your app is complete
- your design looks now like this:

|sign-up|sign-in|
|-|-|
|Screenshot_20221105_225549|Screenshot_20221105_225603|

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Roseokpe 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**